src/gstools/covmodel/fit.py

```diff
@@ -460,36 +622,56 @@ def curve(x, arg1, *args):
     return curve
 
 
-def _post_fitting(model, para, popt, anis, is_dir_vario):
+def _post_fitting(model, para, popt, sill, anis, is_dir_vario, sum_cfg):
     """Postprocess fitting results and application to model."""
+    is_sum = hasattr(model, "sub_arg")
     fit_para = {}
     para_skip = 0
-    opt_skip = 0
-    var_tmp = 0.0  # init value
-    for par in DEFAULT_PARA:
+    for par in model.iso_arg:
         if para[par]:
-            if par == "var":  # set variance last
-                var_tmp = popt[para_skip]
-            else:
-                setattr(model, par, popt[para_skip])
+            setattr(model, par, popt[para_skip])
             fit_para[par] = popt[para_skip]
             para_skip += 1
         else:
             fit_para[par] = getattr(model, par)
-    for opt in model.opt_arg:
-        if para[opt]:
-            setattr(model, opt, popt[para_skip + opt_skip])
-            fit_para[opt] = popt[para_skip + opt_skip]
-            opt_skip += 1
-        else:
-            fit_para[opt] = getattr(model, opt)
+    # set var and len-scale ratios in sum-models
+    if is_sum:
+        if sum_cfg["var_size"] > 0:
+            var_vals = popt[para_skip : para_skip + sum_cfg["var_size"]]
+            para_skip += sum_cfg["var_size"]
+            if sum_cfg["var_fix"]:
+                model.set_var_weights(
+                    stick_breaking_uniform(var_vals),
+                    sum_cfg["var_skip"],
+                    sum_cfg["fix"]["var"],
+                )
+            else:
+                for i, val in zip(sum_cfg["var_fit"], var_vals):
+                    setattr(model, f"var_{i}", val)
+        if sum_cfg["len_size"] > 0:
+            len_vals = popt[para_skip : para_skip + sum_cfg["len_size"]]
+            para_skip += sum_cfg["len_size"]
+            if sum_cfg["len_fix"]:
+                model.set_len_weights(
+                    stick_breaking_uniform(len_vals),
+                    sum_cfg["len_skip"],
+                    sum_cfg["fix"]["len_scale"],
+                )
+            else:
+                for i, val in zip(sum_cfg["len_fit"], len_vals):
+                    setattr(model, f"len_scale_{i}", val)
+        for i in range(model.size):
+            fit_para[f"var_{i}"] = model.vars[i]
+            fit_para[f"len_scale_{i}"] = model.len_scales[i]
+    # handle sill
+    if sill is not None and para["var"]:
+        nugget = sill - model.var
+        fit_para["nugget"] = nugget
+        model.nugget = nugget
```

Comment:
There are quite a few pretty large `if is_sum` branches all over the place here. Maybe you can find a better structure for the code in this module?

Reply:
I think this is a task for the future to rework the fitting in general. I think a class for a variogram fitter could be helpful to not pass around all the the information over and over again.

For now I would keep it like this.

Reply:
See #382